### ISSUE TITLE
Glide not installing all dependencies of Gin (Godep based)

### ISSUE THREAD

**Issue by robvdl** [Masterminds/glide#525]
If I create a simple glide.yaml file that imports Gin;

import:
  - package: github.com/gin-gonic/gin

Then do a "glide up", compile my app, I get the error that two of the dependencies listed in Gin's Godep.json file lists are not installed:

github.com/golang/protobuf/proto and gopkg.in/go-playground/validator.v8

These packages are clearly listed in the Godeps/Godeps.json file in the GIN repository, so why then if I include Gin in my poroject and do a "glide up" do I not get these two packages installed.  I thought Glide could handle Godeps projects, but it doesn't look like it is working for Gin, is this a bug? or am I doing something wrong?

It appears I need to re-list the dependencies that Gin lists in it's Godeps.json file, this seems wrong to me.

**Comment by jonbo** [comment]
I ran into the exact issue myself with Gin. It appears Glide fails to retrieve dependencies of the dependency (Gin, in this case) if it wasn't already imported in your project at the time of first installing it with 'glide up'. 

A follow-up 'glide up' should fix it (assuming you have it imported now) and retrieve those missing dependencies.

(This was my issue/solution anyway.)

**Comment by marioluan** [comment]
Same here. I had to add gin's dependencies to my own project in order to make it work.

**Comment by robvdl** [comment]
Ah thank you for this, so it definitely is a bug, it isn't just me :) also good there is a workaround for now.

**Comment by mattfarina** [comment]
Using the tip of master I cannot reproduce this problem with the following:

**main.go:**
```go
package main

import (
	"fmt"

	_ "github.com/gin-gonic/gin"
)

func main() {
	fmt.Println("test")
}
```

**glide.yaml:**
```yaml
package: foo
import:
- package: github.com/gin-gonic/gin
```

**glide.lock:**
```yaml
hash: 1b9c474483dd1c98724d72487cacd0dc3fa6cd0f876bb422b44779250d66b6f3
updated: 2016-08-30T20:59:16.182465175-04:00
imports:
- name: github.com/gin-gonic/gin
  version: f931d1ea80ae95a6fc739213cdd9399bd2967fb6
  subpackages:
  - binding
  - render
- name: github.com/golang/protobuf
  version: 2402d76f3d41f928c7902a765dfc872356dd3aad
  subpackages:
  - proto
- name: github.com/manucorporat/sse
  version: ee05b128a739a0fb76c7ebd3ae4810c1de808d6d
- name: golang.org/x/net
  version: f315505cf3349909cdf013ea56690da34e96a451
  subpackages:
  - context
- name: gopkg.in/go-playground/validator.v8
  version: c193cecd124b5cc722d7ee5538e945bdb3348435
- name: gopkg.in/yaml.v2
  version: e4d366fc3c7938e2958e662b4258c7a89e1f0e3e
testImports: []
```

You'll notice both `github.com/golang/protobuf/proto` and `gopkg.in/go-playground/validator.v8` are present and they are in the `vendor/` directory.

Can you provide me a way to reproduce this?

**Comment by robvdl** [comment]
Do you have to manually create the glide.lock file, because I don't seem to have this file.

Anyway, I am on Ubuntu 14.04, Go 1.7

What I did:

* Create an empty foo package with main.go
* glide init
* Add the github.com/gin-gonic dependency to glide.yaml
* glide up

What I expected to see:

* vendor directory contains gin and all it's dependencies

What I actually see:

* vendor directory only contains gin, nothing else

I also tried a second glide up as stated by jonob but it doesn't fetch Gin's dependencies like he experienced.

**Comment by robvdl** [comment]
Interestingly, typing "glide guess" does list the dependencies of Gin, but glide up doesn't fetch them.

**Comment by jonbo** [comment]
@robvdl Do you have gin imported in your main.go? That's what fixed it for me. Then 'glide up' worked. (Note: I haven't tested any recent builds though.)

**Comment by robvdl** [comment]
@jonob yes I did, I used the main.go sample posted here

**Comment by franciscocpg** [comment]
@robvdl, I can confirm what @jonbo said using glide version v0.12.1, go 1.6.2 and ubuntu 16.04.
Which glide version are you using?

here is my tests results
with empty **main.go**
```
glide up
[INFO]	Downloading dependencies. Please wait...
[INFO]	--> Fetching updates for github.com/gin-gonic/gin.
[INFO]	Resolving imports
[INFO]	Downloading dependencies. Please wait...
[INFO]	Setting references for remaining imports
[INFO]	Exporting resolved dependencies...
[INFO]	--> Exporting github.com/gin-gonic/gin
[INFO]	Replacing existing vendor dependencies
[INFO]	Project relies on 1 dependencies.

ls vendor 
github.com
```
with @mattfarina suggested **main.go**
```
glide up
[INFO]	Downloading dependencies. Please wait...
[INFO]	--> Fetching updates for github.com/gin-gonic/gin.
[INFO]	Resolving imports
[INFO]	Found Godeps.json file in /home/francisco/programas/glide/cache/src/https-github.com-gin-gonic-gin
[INFO]	--> Parsing Godeps metadata...
[INFO]	--> Fetching updates for github.com/manucorporat/sse.
[INFO]	--> Setting version for github.com/manucorporat/sse to ee05b128a739a0fb76c7ebd3ae4810c1de808d6d.
[INFO]	--> Fetching updates for golang.org/x/net.
[INFO]	--> Setting version for golang.org/x/net to f315505cf3349909cdf013ea56690da34e96a451.
[INFO]	--> Fetching updates for github.com/golang/protobuf.
[INFO]	--> Setting version for github.com/golang/protobuf to 2402d76f3d41f928c7902a765dfc872356dd3aad.
[INFO]	--> Fetching updates for gopkg.in/go-playground/validator.v8.
[INFO]	--> Setting version for gopkg.in/go-playground/validator.v8 to c193cecd124b5cc722d7ee5538e945bdb3348435.
[INFO]	--> Fetching updates for gopkg.in/yaml.v2.
[INFO]	Downloading dependencies. Please wait...
[INFO]	Setting references for remaining imports
[INFO]	Exporting resolved dependencies...
[INFO]	--> Exporting github.com/gin-gonic/gin
[INFO]	--> Exporting github.com/manucorporat/sse
[INFO]	--> Exporting github.com/golang/protobuf
[INFO]	--> Exporting golang.org/x/net
[INFO]	--> Exporting gopkg.in/yaml.v2
[INFO]	--> Exporting gopkg.in/go-playground/validator.v8
[INFO]	Replacing existing vendor dependencies
[INFO]	Project relies on 6 dependencies.

ls vendor
github.com  golang.org  gopkg.in
```

So it's working fine.

**Comment by robvdl** [comment]
I found the issue, my apologies, I am closing this.

Although I somehow had the source of glide 0.12.1... for some reason I still had the binary of some older "glide dev" version. How did it not end up compiling the glide binary? I don't know but I have a feeling it was because I used Go-Package-Store to update things.

I noticed because my glide up output was different, even though the glide CHANGELOG.md read 0.12.1 so something was "off"

I reinstalled glide using go get -u and things are working, thanks heaps, and maybe it was an issue in Go-Package-Store (which I should really stop using anyway) but it's definitely not a glide issue.
